internal/contentmanager/cache.go

```diff
@@ -0,0 +1,115 @@
+package contentmanager
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Cache is a storage mechanism for keeping track of
+// managed content sources. It also exposes methods
+// for retrieving the stored references of the managed
+// content
+type Cache interface {
+	// Get attempts to fetch the object with the provided namespace/name and GroupVersionKind
+	Get(client.ObjectKey, schema.GroupVersionKind) (runtime.Object, error)
+	// List attempts to list all the objects with the provided GroupVersionKind and label selector
+	List(schema.GroupVersionKind, labels.Selector) ([]runtime.Object, error)
+	// Close stops all sources belonging to the cache
+	Close()
+	// AddSource adds a new ReaderCloserSyncincSource to the cache for the provided GroupVersionKind
+	AddSource(schema.GroupVersionKind, ReaderCloserSyncingSource)
+	// RemoveSource stops and removes a source from cache for the provided GroupVersionKind
+	RemoveSource(schema.GroupVersionKind)
+	// Sources returns the current mapping of GroupVersionKind to ReaderCloserSyncingSource
+	Sources() map[schema.GroupVersionKind]ReaderCloserSyncingSource
+}
+
+// ReaderCloserSyncingSource is a wrapper of the controller-runtime
+// source.SyncingSource that includes methods for:
+//   - Getting a specific resource synced by the source
+//   - Listing resources synced by the source
+//   - Closing the source, stopping it's interaction with the Kubernetes API server and reaction to events
+type ReaderCloserSyncingSource interface {
+	source.SyncingSource
+	// Get attempts to fetch the object with the provided namespace/name
+	Get(client.ObjectKey) (runtime.Object, error)
+	// List attempts to list all the objects with the provided label selector
+	List(labels.Selector) ([]runtime.Object, error)
```

Comment:
Get rid of these as well since this was added to facilitate the higher level `Cache.Get()` and `Cache.List()` functions